src/navigator/RewardsPill.tsx

```diff
@@ -2,29 +2,39 @@ import React from 'react'
 import { useTranslation } from 'react-i18next'
 import { RewardsEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
-import { rewardsEnabledSelector } from 'src/app/selectors'
+import { phoneNumberVerifiedSelector, rewardsEnabledSelector } from 'src/app/selectors'
 import Pill from 'src/components/Pill'
 import { isE2EEnv } from 'src/config'
 import { RewardsScreenOrigin } from 'src/consumerIncentives/analyticsEventsTracker'
+import { superchargeInfoSelector } from 'src/consumerIncentives/selectors'
 import Rings from 'src/icons/Rings'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import useSelector from 'src/redux/useSelector'
+import { getFeatureGate } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
 
 function RewardsPill() {
   const { t } = useTranslation()
 
+  const rewardsEnabled = useSelector(rewardsEnabledSelector)
+  const phoneNumberVerified = useSelector(phoneNumberVerifiedSelector)
+  const { hasBalanceForSupercharge } = useSelector(superchargeInfoSelector)
+  const restrictSuperchargeForClaimOnly = getFeatureGate(
+    StatsigFeatureGates.RESTRICT_SUPERCHARGE_FOR_CLAIM_ONLY
+  )
+  const isSupercharging = phoneNumberVerified && hasBalanceForSupercharge
+
   const onOpenRewards = () => {
     navigate(Screens.ConsumerIncentivesHomeScreen)
     ValoraAnalytics.track(RewardsEvents.rewards_screen_opened, {
       origin: RewardsScreenOrigin.RewardPill,
     })
   }
 
-  const rewardsEnabled = useSelector(rewardsEnabledSelector)
   const showRewardsPill = isE2EEnv || rewardsEnabled
 
-  if (!showRewardsPill) {
+  if ((restrictSuperchargeForClaimOnly && !isSupercharging) || !showRewardsPill) {
```

Comment:
I would prefer to combine all conditions under `showRewardsPill` as its name suggests, but it is not a strong opinion.

```suggestion
  const isRestricted = restrictSuperchargeForClaimOnly && !isSupercharging
  const isEnabled = rewardsEnabled && !isRestricted
  const showRewardsPill =  isE2Env || isEnabled

  if (!showRewardsPill) {
     return null
  }
```

Reply:
thanks, i think you commented on an outdated version of the code. what do you think of it now?

Reply:
Thanks! I think now it's much cleaner, all this complexity is in one place.